lib/block/block.go

```diff
@@ -23,6 +24,7 @@ const (
 type Block struct {
 	Header
 	Transactions []string `json:"transactions"` /* []Transaction.GetHash() */
+	Transaction  string   `json:"transaction"`  /* ProposerTransaction */
```

Comment:
I figure, `Transaction` is too general word.
And `ProposerTransaction` is not well fit for this purpose I think.
How about `SpecialTransaction` or `ExceptionalTransaction`?

Reply:
@Charleslee522 'Special' and 'Exceptional' looks also too general :) I had 2 candidates,

* the current one, 'ProposerTransaction'
* and dismissed one, 'BlockTransaction'

'BlockTransaction' is already used for `block.BlockTransaction`, so ...  This transaction is proposed by proposer and it will contain the other kind of Operation, like for #461 .

Please suggest me the other wonderful name for it.

Reply:
> > And is it possible to store block with only transactions? not add one more transaction?
I think BlockTransaction should store Transaction, I can not imagine the other kind of transaction for BlockTransaction at this time.

I mean this code.
Why not store the `Transaction` onto the first position of `Transactions`?
@spikeekips

Reply:
@kfangw You pointed very sharply. The first design of this feature was that. Instead of adding new field for `ProposerTransaction`, set it at the first item in `Ballot.B.Transactions`. But I dropped this idea because of broadcasting problem. I want `ProposerTransaction` should be validated without broadcasting. Without broadcasting, it should be some where in `Ballot`.

Reply:
Ok. understood. thanks for your kind explanation.
@spikeekips

Reply:
@spikeekips 
> I figure, Transaction is too general word.

It means I think 
```
type Block struct {
	Header
	Transactions []string `json:"transactions"` /* []Transaction.GetHash() */
	ProposerTransaction  string   `json:"transaction"`
}
```
or
```
type Block struct {
	Header
	Txs []string `json:"transactions"` /* []Transaction.GetHash() */
	ProposerTx  string   `json:"transaction"`
}
```
is better than
```
type Block struct {
	Header
	Transactions []string `json:"transactions"` /* []Transaction.GetHash() */
	Transaction  string   `json:"transaction"`  /* ProposerTransaction */
}
```

Reply:
@spikeekips And normal transactions are also proposed by client.
So I thought that is general.
What the `ProposerTransaction` means is "Transaction by Proposer", right?

Reply:
Right.

Reply:
@spikeekips What do you think about this? https://github.com/bosnet/sebak/pull/469/files#r221781065